code/starter_script - Copy.R

```diff
@@ -0,0 +1,64 @@
+# an example of paired predictions and observations
+library(dplyr)
+library(plyr)
+dat <- readRDS('data/predicted_observed_temps.rds')
+head(dat)
+
+# note that SNTemp predictions (uncalibrated process model)
+# have predictions for all places/times
+# here you just have the test period from the hybrid model (rgcn2_full_temp_c)
+# and sparse observations (some reaches measured, some not)
+
+summary(dat)
+str(dat)
+# example RMSE calculation
+# https://github.com/USGS-R/delaware-model-prep/blob/master/3_predictions/src/munge_predictions.R#L103-L113
+
+# removing rows with na values. 
+rgnc_dat_filter <- dat %>% filter(!is.na(temp_c), !is.na(rgcn2_full_temp_c))
+str(rgnc_dat_filter)
+summary(rgnc_dat_filter)
+
+## Absolute Residual Error: calculating Mean Absolute Error (MAE) metric to compare both models prediction data vs observed data. We found the mae by finding the sum of the absolute value difference in predicted and observed temperature. Then divided the sum by the n row in grouped data.
+
+cal_mae <- function(observe_data, predict_data) {
+  rgnc_dat_filter %>%
+    mutate( diff_observed_predicted = observe_data - predict_data ) %>%
+    group_by(seg_id_nat) %>%
+    summarize( mae = round( mean( abs( diff_observed_predicted), na.rm = TRUE ), 2) )
+}
+
+mae_process_model <- cal_mae(rgnc_dat_filter$temp_c, rgnc_dat_filter$sntemp_temp_c)
+mae_hybrid_model <- cal_mae(rgnc_dat_filter$temp_c, rgnc_dat_filter$rgcn2_full_temp_c)
+mae <- left_join( mae_process_model, mae_hybrid_model, by = "seg_id_nat")
+colnames(mae)[2] <- "MAE_Process_Model"
+colnames(mae)[3] <- "MAE_Hybrid_Model"
+# Absolute Residual Error: Root Mean Square Error (RMSE) square the difference in predicted and observed temperature.  sum the squared values, then divided by n of row. Finally,took the square root
+cal_rmse <- function(observe_data, predict_data){
+  rgnc_dat_filter %>%
+    mutate( diff_observed_predicted = observe_data - predict_data ) %>%
+    group_by(seg_id_nat) %>%
+    summarize( rmse = round( sqrt( mean( diff_observed_predicted ^2, na.rm = TRUE ) ) , 2)  )
```

Comment:
Some general style feedback that doesn't impact _what_ the code is doing, but does make the code more readable and standardized across our group. Don't include spaces after a `(` or before `)` within a function. Generally if you have a nested function where the line ends in `))`, there is no space between.  I would write the summarize line as:

`summarize(rmse = round(sqrt(mean(diff_observed_predicted ^ 2, na.rm = TRUE )) , 2))`

I think we largely follow [Hadley Wickam's style guide](http://adv-r.had.co.nz/Style.html).